### ISSUE TITLE
Feature Request: Spell checking 

### ISSUE THREAD

**Issue by drupalshift** [fabiospampinato/notable#108]
Support for spell checking, there are several packages that can implement spell checking in electron, It would be really helpful

**Comment by johnletey** [comment]
Do you know if there's a showdown plugin for spell checking? That would be the easiest for implementing.

**Comment by drupalshift** [comment]
@johnletey https://www.npmjs.com/package/electron-spellchecker
it has auto language detection and right click support

**Comment by InnovativeInventor** [comment]
I think integration with LanguageTool for grammar and spelling would be useful.

**Comment by tiotrom** [comment]
I miss this feature :P - a lot :D

**Comment by OnyxDelta** [comment]
I have been trying to look for plugins that may enable something like this.  I found these two which also provide linting for grammar. However, I am slightly worried about their dependencies being too numerous:
[Proofreader](https://github.com/anthonyjclark/Proofreader)
[Textlint-rule-ginger](https://github.com/textlint-rule/textlint-rule-ginger)
Another option I found was the spellchecker mentioned before:
[electron-spellchecker](https://github.com/electron-userland/electron-spellchecker)
This would be extremely helpful, but I also understand that this may be difficult to implement.

**Comment by ChuckJonas** [comment]
I'm horible at spelling!  Please impliment this feeture as soon as posible

**Comment by blankwall** [comment]
Bump for this feature

**Comment by asmith-holvi** [comment]
Bump.. I have been liking this app, but about to move away since my spelling is not optimal.

**Comment by Grayson112233** [comment]
Bump, this is a deal breaker for me, otherwise an amazing application.

**Comment by alpha-beta-soup** [comment]
One reason why it's important is to support text search. If I misspell some key, long, technical term in a document then it's hard to find it later with search. Text-based search relies on correct spelling.

**Comment by gayanhewa** [comment]
Is this likely to happen?

**Comment by fabiospampinato** [comment]
@gayanhewa Yes, I'm not sure about when though.

**Comment by gayanhewa** [comment]
No worries, I guess if we can work out a way to get 3rd party tools like Grammarly to work that would be more than sufficient.

**Comment by fabiospampinato** [comment]
@gayanhewa You can edit your notes with external editors if you wish, I supposed that would be a somewhat usable workaround until this gets implemented.

**Comment by gayanhewa** [comment]
Yes, it's not a deal-breaker. Just a convenience, especially when writing blog posts and stuff. Internal notes it rarely matters.

**Comment by thwaller** [comment]
I have a specific use for which I like Notable. I tend to write a lot of letters/messages/emails or what ever you wish to call them over time. Although it might seem like an odd use of this software, it works perfectly... except for this.

**Comment by fabiospampinato** [comment]
@thwaller you might want to take a look at [Markdown Here](https://markdown-here.com/), I've been thinking about making a Notable version of that in the future.

**Comment by thwaller** [comment]
@fabiospampinato thanks for the suggestion. I will look further at it, but I am unsure if I would like something that is tied to a browser.

**Comment by epmoyer** [comment]
-  Does the absence of spelling make me abandon the tool?
   - No; I am "all in" and I'm determined to make Notable work for me. :heart: Love it.
- Do I want the feature?
   - Yes, more than *any other feature*. 
      -  Why? It is the *single* missing feature which causes me to use another tool in my workflow.  I am forced to compose in VSCode because I need a spell checker watching my back.  That makes Notable my "note browser" and VSCode my "note composer" until spelling arrives.
   - Would I pay for it?
      - At the very least I'll toss my $5/mo into the pot. I'm now a sponsor at https://github.com/sponsors/fabiospampinato.

**Comment by fabiospampinato** [comment]
@epmoyer 😄 thanks for your support!

This will have to wait after the v1.9 release at least, then I'll look again into what would it take to add support for spell checking, I've just taken a quick look at this again and it seems somewhat complicated 🤔

**Comment by tehp** [comment]
I am going to work on adding this later today. I will report back any progress that I make.

**Comment by fabiospampinato** [comment]
@tehp 👍 If you can add this to [v1.5.1](https://github.com/notable/notable/tree/v1.5.1) I should pretty much be able to port your commits to master.

**Comment by fabiospampinato** [comment]
For future reference: adding support for adding custom works to the dictionary would be useful.

**Comment by Fleisch** [comment]
Seems like you are making good progress. Just giving you another vote that I am excited for the feature!

**Comment by mikejones3** [comment]
Please make this a thing. I want to be able to export notes to my team.

**Comment by hypergig** [comment]
Maybe place a bounty on this feature? I think many of us would be willing to pay to see this happen.

**Comment by bkayes** [comment]
Bump. This feature is critical for me, the app is almost unusable without it

**Comment by dipernaa** [comment]
Any updates on this?

**Comment by fabiospampinato** [comment]
> Any updates on this?

@dipernaa I implemented it some weeks ago, there's a preview video about it in the [chat](https://chat.notable.app), it will be available starting from v1.9 alpha 20 to be released within days here (optimistically tomorrow, it'll probably take a bit longer though): https://github.com/notable/notable-experimental/releases

**Comment by maximilianreimer** [comment]
I also really would like to see this feature being implemented!

**Comment by mikejones3** [comment]
@maximilianreimer it the latest alpha or beta spell check is working! So its almost there!

**Comment by AltamimiYasser** [comment]
Thank you so much, guys! Can't wait for it to be released!!

**Comment by sergeevabc** [comment]
Err… How do I disable spell-checking? My native language is not English, so the content is annoyingly underlined in red.

**Comment by MMK21Hub** [comment]
@sergeevabc Press F8 is the default keybind to toggle spellcheck, or you can use the command pallette.

You'll also find an option to change the spellcheck language, which you may find useful.

**Comment by TheDevelolper** [comment]
I'm using Version 1.8.4 and I can't seem to get spell check enabled on Windows? It just doesn't seem to work. I also installed it for a friend and he can't get it to work either?

**Comment by MMK21Hub** [comment]
@TheDevelolper You need to install the ["insiders"](https://github.com/notable/notable-insiders) version of Notable for the latest features :)

The latest release is beta.10, and it can be downloaded from https://github.com/notable/notable-insiders/releases/tag/v1.9.0-beta.10
